velox/expression/Expr.cpp

```diff
@@ -259,6 +289,110 @@ bool isFlat(const BaseVector& vector) {
 
 } // namespace
 
+bool MutableRemainingRows::deselectNonErrorNulls(
+    const VectorPtr& result,
+    EvalCtx::ErrorVectorPtr& errors) {
+  const uint64_t* flatNulls = nullptr;
+  if (result->mayHaveNulls()) {
+    decoded_.get()->decode(*result, rows());
+    flatNulls = decoded_.get()->nulls();
+  }
+  // A null with no error deselects the row and clears a possible error. An
+  // error adds itself to argument errors.
+  if (context_.errors()) {
+    // There are new errors.
+    auto& remainingRows = mutableRows();
+    context_.ensureErrorsVectorSize(*context_.errorsPtr(), rows().end());
+    auto newErrors = context_.errors();
+    if (flatNulls) {
+      // There are both nulls and errors. Only a null with no error removes a
+      // row.
+      auto errorNulls = newErrors->rawNulls();
+      auto rowBits = mutableRows().asMutableRange().bits();
+      auto nwords = bits::nwords(rows().end());
+      for (auto i = 0; i < nwords; ++i) {
+        auto nullNoError = errorNulls ? flatNulls[i] | errorNulls[i] : 0;
```

Comment:
should the else condition here have flatNulls[i] instead of a 0 ? that is:

`auto nullNoError = errorNulls ? flatNulls[i] | errorNulls[i] : flatNulls[i];`